### DISCUSSION TITLE
DWIN Update?

### DISCUSSION THREAD

**Discussion by Cal-LF** [Jyers/Marlin#218] [Q&A] [answered]
Should the screen FW be updated with the motherboard FW? I see it is dated recently. Do the new icons update also?

**Comment by brett0712** [comment]
I updated the the firmware and LCDs on all of my Ender 3 V2s today for the same reason. Can't say that I really noticed anything different. Although I am not looking too closely. Certainly doesn't seem to have hurt anything.

**Comment by t3chguy** [comment]
I believe there's a config flag you need to enable for if you have updated your LCD to enable the extended icon set

**Comment by brett0712** [comment]
Is it this:

#define CREALITY_DWIN_EXTUI
#if ENABLED(CREALITY_DWIN_EXTUI)
  //
  // Enable custom icons
  // NB: Requires Ender-3 v2 OEM display firmware update, or you will get blank icons!
  //
  //#define CREALITY_DWIN_EXTUI_CUSTOM_ICONS

If so, is what Jyers is using for updated icons considered custom?

**Reply by swbog** [reply]
I think that’s it. What folder is that sauce in

**Reply by brett0712** [reply]
Its in the configuration.h file. Starts around line 2260 - the section is titled "Extensible UI Displays" I'm using one of the configuration.h files from the E2V2 Templates. Marlin-Ender_3_V2_Extensible_UI\Configuration Files\E3V2 Templates

**Comment by swbog** [comment]
Do the items need to be taken out of the folder for the update to take? I saw no color screens

**Reply by brett0712** [reply]
You need to put the DWIN_SET folder on a SD card, remove the back cover of the LCD screen and insert the SD card into the SD card port. Power on the machine and the screen should go from black to blue to orange. Once its orange you can power off the machine, remove the SD card, and power on the machine to confirm a successful flash.

However, as noted above, it seems as though there is a config flag that needs to be uncommented for the new icon set to take/work.

**Comment by Schnabulation** [comment, marked as answer]
At the moment `CREALITY_DWIN_EXTUI_CUSTOM_ICONS` only enables some small icon-changes like the checkbox. All other icons stay the same.

You can test this by enabling `CREALITY_DWIN_EXTUI_CUSTOM_ICONS` but not flashing the new DWIN firmware - all icons that would get replaced will stay empty.

**Reply by Cal-LF** [reply]
Ok, that is what I wanted to know. Thanks.

**Comment by embraceentropy** [comment]
To resurrect this:

This is my current configuration.h information. To enable the customer icons I would need to un-comment //#define CREALITY_DWIN_EXTUI_CUSTOM_ICONS to be #define CREALITY_DWIN_EXTUI_CUSTOM_ICONS, but leave the rest of the code the same? 



#define CREALITY_DWIN_EXTUI
#if ENABLED(CREALITY_DWIN_EXTUI)
  //
  // Enable custom icons
  // NB: Requires Ender-3 v2 OEM display firmware update, or you will get blank icons!
  //
  //#define CREALITY_DWIN_EXTUI_CUSTOM_ICONS
#endif

**Comment by Cal-LF** [comment]
There are new icon packs to try in the Display Firmware folder.
I think the #define CREALITY_DWIN_EXTUI_CUSTOM_ICONS has to be turned on in the Configuration.h to use the custom icon packs.
